<launch>
    <arg name="map_frame" default="map" />
    <arg name="odom_frame" default="odom" />

    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="yaw" default="0.0" />

    <arg name="use_sim_time" default="false" />

    <arg name="use_commander" default="true" />
    <let name="sc_prefix" value="" />
    <group scoped="false" if="$(var use_commander)">
        <let name="sc_prefix" value="sc_" />
    </group>

    <node name="set_goal" pkg="nav2_oneshot_nodes" exec="$(var sc_prefix)set_goal" output="screen">
        <param name="map_frame" value="$(var map_frame)" />
        <param name="odom_frame" value="$(var odom_frame)" />
        <param name="x" value="$(var x)" />
        <param name="y" value="$(var y)" />
        <param name="yaw" value="$(var yaw)" />
        <param name="use_sim_time" value="$(var use_sim_time)" />
        <remap from="goal_pose" to="/goal_pose" />
        <remap from="goal_status" to="/navigate_to_pose/_action/status" />
    </node>
</launch>